org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```diff
@@ -205,10 +206,29 @@ private static void installBundle(BundleContext context, Set<Bundle> bundlesToSt
 	 * @throws BundleException
 	 */
 	private static void uninstallBundle(Set<Bundle> bundlesToStart, Set<Bundle> toRefresh, Bundle bundle) throws BundleException {
-		bundle.uninstall();
-		JavaLanguageServerPlugin.logInfo("Uninstalled " + bundle.getLocation());
-		toRefresh.add(bundle);
-		bundlesToStart.remove(bundle);
+		if (selfExcludedFromDependencyClosure(bundle)) {
+			bundle.uninstall();
+			JavaLanguageServerPlugin.logInfo("Uninstalled " + bundle.getLocation());
+			toRefresh.add(bundle);
+			bundlesToStart.remove(bundle);
+		}
+	}
+
+	private static boolean selfExcludedFromDependencyClosure(Bundle bundle) throws BundleException {
+		Collection<Bundle> bundles = getFrameworkWiring().getDependencyClosure(Collections.singletonList(bundle));
+		for (Bundle b : bundles) {
+			if (IConstants.PLUGIN_ID.equals(b.getSymbolicName())) {
+				JavaLanguageServerPlugin.logError("Cannot refresh the " + IConstants.PLUGIN_ID + " bundle.");

```

Comment:
Could we update this message to be :

`"Cannot refresh bundle " + b.getSymbolicName() + " because its dependency closure includes the " + IConstants.PLUGIN_ID + " bundle."`

Reply:
Fixed.